services/gateway/client/client.go

```diff
@@ -186,6 +186,14 @@ func (c *Client) DeleteProjectGroupSecret(ctx context.Context, projectGroupRef,
 	return c.getResponse(ctx, "DELETE", path.Join("/projectgroups", url.PathEscape(projectGroupRef), "secrets", secretName), nil, jsonContent, nil)
 }
 
+func (c *Client) GetProjectGroupSecrets(ctx context.Context, projectRef string) ([]*gwapitypes.SecretResponse, *http.Response, error) {
```

Comment:
You should add two input parameter: `tree bool` and `removeoverridden bool` and set them in the query parameter only when true. The same for the other added client functions.